src/docs/arch-hs.dox

```diff
@@ -1,8 +1,272 @@
+    <td style="width: 110.312px; height: 32px; text-align: center;">STANDARD</td>
+    <td style="width: 148.141px; height: 32px; text-align: center;">Value from U3</td>
+  </tr>
+  <tr style="height: 32px;">
+    <td style="width: 72.5156px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 89.875px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 97.0312px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 93.9688px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 92.4375px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 105.719px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 110.312px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+    <td style="width: 148.141px; height: 32px; text-align: center;">
+      <strong>...</strong>
+    </td>
+  </tr>
+</table>
+
+For the history store table entry corresponding to the update \c U1, the start timestamp in the key
+is the start timestamp for the update \c U1 and the stop timestamp in the value is the start
+timestamp for the update \c U2.
+
+When a modified history store table page is processed by reconciliation, a new page image is
+created without records which have globally visible tombstones. This ensures that WiredTiger is
+only keeping the relevant historical data required to serve the oldest reader in the system or
+as dictated by the oldest timestamp set by the application. Once all records on a page are obsolete,
+the page itself can be removed to reduce the size of the history store table
+(see @ref arch-transaction for more details about oldest timestamp).
+
+@section arch_hs_read Searching for older versions of a key in History Store
+When looking for an update visible to the current transaction, WiredTiger first searches the update
+chain for any visible updates. If there is no visible update in the chain, WiredTiger then
+inspects the on-disk version of the key. If that version is not visible to the transaction,
+WiredTiger searches the history store table for a visible update for the key. When
+searching for an update that satisfies the read timestamp constraint, WiredTiger starts
+with the newest update of the key in the history store table and then iterates through the older
+updates until there are no updates left to process. Note that although there can be multiple updates
+in the history store for a key and read timestamp combination, each update would have different
+visibility based on transaction and timestamp based visibility rules. In case there are no records
+visible in the history store table, \c WT_NOTFOUND error is returned to the reader.
+
+@section arch_hs_rts History tore and rollback-to-stable
```

Comment:
Minor typo:  History tore ->  History store

Reply:
Fixed the typo.